marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/OutputEndpointImpl.java

```diff
@@ -87,17 +87,15 @@ public void awaitCompletion() {
                     throw new RuntimeException("error while calling "+getEndpoint().getEndpointPath(), throwable);
                 }
 
-                InputStream[] result = null;
                 if(output != null) {
-                    result = output.toArray(size -> new InputStream[size]);
-                }
-                if (allowsEndpointState()) {
-                        if(result !=null) {
+                    InputStream[] result = output.toArray(size -> new InputStream[size]);
+
+                    if (allowsEndpointState() && result != null && result.length>0) {
```

Comment:
As I understand Stream, if output is not null, the output.toArray() is guaranteed to be not null.

More generally, either the if test doesn't need to test for result != null or the for loop should have the same test.